### DISCUSSION TITLE
Annotating existing properties as `[Indexed]` vs schema version

### DISCUSSION THREAD

**Discussion by bdach** [realm/realm-dotnet#3370] [Q&A] [answered]
Assume I have the following realm model (simplified):

```csharp
public class ScoreInfo : RealmObject
{
    [PrimaryKey]
    public Guid ID { get; set; }

    public string Hash { get; set; }
}
```

I want to annotate `Hash` with `[Indexed]` for faster lookups. Is it necessary, recommended, or required to increment the `SchemaVersion` in the `RealmConfiguration` after applying such a change?

I tried to look for an answer to this in docs and only found indexed fields mentioned [here](https://www.mongodb.com/docs/realm/sdk/dotnet/model-data/define-object-model/#std-label-dotnet-index-field) which is not explicit on the matter at hand. The docs on [changing schema](https://www.mongodb.com/docs/realm/sdk/dotnet/model-data/change-an-object-model/#schema-version) also do not mention index annotations in any way.

**Comment by nirinchev** [comment, marked as answer]
Adding and removing indexes does not require bumping the schema version as it doesn't necessitate migrations (the data is still the same, we just build an index for faster lookups).

**Reply by bdach** [reply]
Thanks for the prompt reply!

**Reply by peppy** [reply]
Is there any documentation as to when the index is built (and if removing the `indexed` attribute, when/how it is removed)?

**Reply by nirinchev** [reply]
We don't have docs for this as far as I know. But the changes are applied synchronously when the realm is opened.
